pkg/tso/keyspace_group_manager.go

```diff
@@ -1116,9 +1141,18 @@ func (kgm *KeyspaceGroupManager) finishSplitKeyspaceGroup(id uint32) error {
 			zap.Int("status-code", statusCode))
 		return errs.ErrSendRequest.FastGenByArgs()
 	}
-	// Pre-update the split keyspace group split state in memory.
-	splitGroup.SplitState = nil
-	kgm.kgs[id] = splitGroup
+	// Pre-update the split keyspace group's split state in memory.
+	// Note: to avoid data race with state read APIs, we always replace the group in memory as a whole.
+	newSplitGroup := endpoint.NewKeyspaceGroup(
+		splitGroup.ID,
+		splitGroup.UserKind,
+		nil,
+		splitGroup.MergeState,
+		splitGroup.Members,
+		splitGroup.Keyspaces,
```

Comment:
It will also meet race when other read this params?

Reply:
Added comment to explain. 

	// Pre-update the merge target keyspace group's merge state in memory.
	// Note: to avoid data race with state read APIs, we always replace the group in memory as a whole.
	**// For now, we only have scenarios to update split state/merge state, and the other fields are always**
	**// loaded from etcd without any modification, so we can simply copy the group and replace the state.**